### ISSUE TITLE
Lp regularizers have no hyperparameters?

### ISSUE THREAD

**Issue by RobinAlgayres** [KevinMusgrave/pytorch-metric-learning#299]
Hi,
To my understanding, LpRegularizer(p=2) is equivalent to the use of 'weight decay' in my pytorch optimizer. However, in pytorch weight decay uses an float parameter to tune the strenght of the regularization, but Lpregularizer does not. Does it mean that LpRegularizer is not the same as the pytorch weight decay? 
Thank you for you time

**Comment by KevinMusgrave** [comment]
You can set ```embedding_reg_weight```:

```python
from pytorch_metric_learning import losses, regularizers

r = regularizers.LpRegularizer()

loss_fn = losses.ContrastiveLoss(embedding_regularizer=r, embedding_reg_weight=0.1)
```

That said, it still won't be equivalent to weight decay because I think weight decay uses the **squared** L2 norm. 

Perhaps I should add an input parameter ```power``` to control that, so that if you set ```power=2``` then it will square the L2 norm.

**Comment by KevinMusgrave** [comment]
Even if the norm is squared, the value might be off by a factor of 2 when compared with weight decay, though I'm not entirely sure. See [this](https://bbabenko.github.io/weight-decay/) for more.

**Comment by RobinAlgayres** [comment]
Well it seems there is already a power factor in the LpRegularizer

regularizers.LpRegularizer(p=2, **kwargs)

So I suppose if I use embedding_reg_weight in my loss, it boils down to roughly the same thing. 
Thank you for your help!

**Comment by KevinMusgrave** [comment]
Actually ```p``` represents the type of norm. 

```p=2```  == L2 distance, ```p=1``` == L1 distance, etc

**Comment by RobinAlgayres** [comment]
oh my bad, yes sure.
So indeed there might be a missing factor of 2 somewhere. But it's not a problem in my situation. 
Thank you!
